sdk/src/main/java/com/kin/ecosystem/Kin.java

```diff
@@ -197,37 +186,20 @@ public static void login(@NonNull WhitelistData whitelistData, KinCallback<Void>
 	 * @param loginCallback a login callback whether login succeed or not.
 	 */
 	public static void login(@NonNull String jwt, KinCallback<Void> loginCallback) {
-		SignInData signInData = getJwtSignInData(jwt);
-		internalLogin(signInData, loginCallback);
+		internalLogin(jwt, loginCallback);
 	}
 
-	private static SignInData getWhiteListSignInData(@NonNull final WhitelistData whitelistData) {
-		return new SignInData()
-			.signInType(SignInTypeEnum.WHITELIST)
-			.userId(whitelistData.getUserID())
-			.appId(whitelistData.getAppID())
-			.apiKey(whitelistData.getApiKey());
-	}
-
-	private static SignInData getJwtSignInData(@NonNull final String jwt) {
-		return new SignInData()
-			.signInType(SignInTypeEnum.JWT)
-			.jwt(jwt);
-	}
-
-	private static void internalLogin(@NonNull final SignInData signInData, final KinCallback<Void> loginCallback) {
+	private static void internalLogin(@NonNull final String jwt, final KinCallback<Void> loginCallback) {
 
 		try {
 			checkInstanceNotNull();
+			AuthRepository.getInstance().setJWT(jwt);
 		} catch (final ClientException exception) {
-			instance.executorsUtil.mainThread().execute(new Runnable() {
-				@Override
-				public void run() {
-					loginCallback.onFailure(exception);
-				}
-			});
+			sendLoginFailed(exception, loginCallback);
+		} catch (JSONException exception) {
+			sendLoginFailed(ErrorUtil.getClientException(ClientException.BAD_CONFIGURATION, exception), loginCallback);
```

Comment:
BAD_CONFIGURATION - should be INVALID_REGISTRATION_JWT?

Reply:
I don't think so, invalid jwt is a detail inside a domain of BAD_CONFIGURATION